jerry-core/parser/regexp/re-compiler.c

```diff
@@ -452,12 +453,12 @@ static const re_compiled_code_t *re_cache[RE_CACHE_SIZE];
  * @return compiled bytecode - if found
  *         NULL              - otherwise
  */
-const re_compiled_code_t *
+static const re_compiled_code_t *
 re_find_bytecode_in_cache (ecma_string_t *pattern_str_p, /**< pattern string */
                            uint16_t flags, /**< flags */
-                           uint32_t *idx) /**< [out] index */
+                           uint16_t *idx) /**< [out] index */
```

Comment:
I've reviewed this function and its call site in `re_compile_bytecode` and I'm wondering whether the complexity of its signature is really needed. Wouldn't 
`static uin16_t re_find_bytecode_in_cache (ecma_string_t *pattern_str_p, uint16_t flags)`
suffice? 

With the following semantics: if the returned value equals `RE_CACHE_SIZE`, then the pattern was not found in the cache and also there is no free slot in the cache; if the returned value is a valid index and the corresponding slot in the cache is `NULL`, then the pattern was not found but index points to a free slot; finally, if the index is valid and the slot is non-NULL, then that's the bytecode we looked for.

This would eliminate the need for the out parameter, i.e., a pointer to a short int, which is quite often dereferenced in a loop in the body of the function.